akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphBalanceSpec.scala

```diff
@@ -127,18 +129,21 @@ class GraphBalanceSpec extends AkkaSpec {
       val numElementsForSink = 10000
       val outputs = Sink.fold[Int, Int](0)(_ + _)
 
-      val (r1, r2, r3) = FlowGraph.closed(outputs, outputs, outputs)(Tuple3.apply) { implicit b ⇒
+      val results = FlowGraph.closed(outputs, outputs, outputs)(List(_, _, _)) { implicit b ⇒
         (o1, o2, o3) ⇒
           val balance = b.add(Balance[Int](3, waitForAllDownstreams = true))
           Source.repeat(1).take(numElementsForSink * 3) ~> balance.in
-          balance.out(0) ~> o1.inlet
-          balance.out(1) ~> o2.inlet
-          balance.out(2) ~> o3.inlet
+          balance.out(0) ~> o1
+          balance.out(1) ~> o2
+          balance.out(2) ~> o3
       }.run()
 
-      Await.result(r1, 3.seconds) should be(numElementsForSink +- 2000)
-      Await.result(r2, 3.seconds) should be(numElementsForSink +- 2000)
-      Await.result(r3, 3.seconds) should be(numElementsForSink +- 2000)
+      import system.dispatcher
+      val sum = Future.sequence(results).map { res ⇒
+        res should not contain 0
```

Comment:
It is. But there is `Await.result` on that future below which will wait for result and therefore execute this assertion.